From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 23 Feb 2015 05:09:41 +0000
Subject: rmap: Fix ABI change in 3.16.7-ckt5
Forwarded: not-needed

'mm: prevent endless growth of anon_vma hierarchy' in 3.16.7-ckt5
added two new members to struct anon_vma.

This structure is always allocated by anon_vma_alloc() so it is OK to
add members, but we need to add them at the end and hide them from
genksyms.

--- a/include/linux/rmap.h
+++ b/include/linux/rmap.h
@@ -37,16 +37,6 @@ struct anon_vma {
 	atomic_t refcount;
 
 	/*
-	 * Count of child anon_vmas and VMAs which points to this anon_vma.
-	 *
-	 * This counter is used for making decision about reusing anon_vma
-	 * instead of forking new one. See comments in function anon_vma_clone.
-	 */
-	unsigned degree;
-
-	struct anon_vma *parent;	/* Parent of this anon_vma */
-
-	/*
 	 * NOTE: the LSB of the rb_root.rb_node is set by
 	 * mm_take_all_locks() _after_ taking the above lock. So the
 	 * rb_root must only be read/written after taking the above lock
@@ -55,6 +45,18 @@ struct anon_vma {
 	 * mm_take_all_locks() (mm_all_locks_mutex).
 	 */
 	struct rb_root rb_root;	/* Interval tree of private "related" vmas */
+
+#ifndef __GENKSYMS__
+	/*
+	 * Count of child anon_vmas and VMAs which points to this anon_vma.
+	 *
+	 * This counter is used for making decision about reusing anon_vma
+	 * instead of forking new one. See comments in function anon_vma_clone.
+	 */
+	unsigned degree;
+
+	struct anon_vma *parent;	/* Parent of this anon_vma */
+#endif
 };
 
 /*
